ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java

```diff
@@ -1262,8 +1262,8 @@ public static boolean isAcid(FileSystem fileSystem, Path directory,
    * @throws IOException on filesystem errors
    */
   public static Directory getAcidState(FileSystem fileSystem, Path candidateDirectory, Configuration conf,
-      ValidWriteIdList writeIdList, Ref<Boolean> useFileIds, boolean ignoreEmptyFiles) throws IOException {
-    return getAcidState(fileSystem, candidateDirectory, conf, writeIdList, useFileIds, ignoreEmptyFiles, null);
+      ValidWriteIdList writeIdList, ValidTxnList validTxnList, Ref<Boolean> useFileIds, boolean ignoreEmptyFiles) throws IOException {
+    return getAcidState(fileSystem, candidateDirectory, conf, writeIdList, validTxnList, useFileIds, ignoreEmptyFiles, null);
```

Comment:
I will create a separate issue to change getAcidState to get just one parameter with builder pattern, because this is getting out of hand

Reply:
Maybe create a separate class for AcidState?